pkg/cmd/cli/kubectl_compat_test.go

```diff
@@ -53,3 +53,29 @@ kubectlLoop:
 	}
 
 }
+
+// this only checks one level deep for nested commands, but it does ensure that we've gotten several
+// --validate flags.  Based on that we can reasonably assume we got them in the kube commands since they
+// all share the same registration.
+func TestValidateDisabled(t *testing.T) {
+	f := clientcmd.New(pflag.NewFlagSet("name", pflag.ContinueOnError))
+
+	oc := NewCommandCLI("oc", "oc", &bytes.Buffer{}, ioutil.Discard, ioutil.Discard)
+	kubectl := kcmd.NewKubectlCommand(f.Factory, nil, ioutil.Discard, ioutil.Discard)
+
+	for _, kubecmd := range kubectl.Commands() {
+		for _, occmd := range oc.Commands() {
+			if kubecmd.Name() == occmd.Name() {
+				ocValidateFlag := occmd.Flags().Lookup("validate")
+				if ocValidateFlag == nil {
+					continue
+				}
+
+				if ocValidateFlag.DefValue != "false" {
```

Comment:
There is no point is checking `DefValue` since it is only for the usage message (https://godoc.org/github.com/spf13/pflag#Flag). You probably want .Value.String().